Added metamask-extension-provider.md to docs (#186)
MetaMask/metamask-docs: docs/guide/metamask-extension-provider.md

# MetaMask Extension Provider

> A module for providing a MetaMask [provider](https://github.com/ethereum/wiki/wiki/JavaScript-API#web3currentprovider) to other WebExtensions.

Reviewer: I think I will adjust the language so the module is documented from its own perspective.
```suggestion
A module for accessing the user's MetaMask [provider](https://github.com/ethereum/wiki/wiki/JavaScript-API#web3currentprovider) from other WebExtensions.
```
